### ISSUE TITLE
RestOnlyClient with Baerer Token Broken

### ISSUE THREAD

**Issue by HurricanKai** [DSharpPlus/DSharpPlus#243]
# Summary
When using the RestOnlyClient with an Baerer Token, it throws 400: Bad Request

# Details
ASP.net / .net Core 2.0, Win10.

Its made shure that the Baerer Token works, because the normal DiscordClient is able to do it.
-> This may also be to the fact that DiscordClient does not trys to Get Guilds on start.

# Steps to reproduce
1. Get a Baerer Token
2. Create a Normal Discord Client -> See it work
3. Create a Rest Only Client -> Get your 400: Bad Request

**Comment by NaamloosDT** [comment]
could you attach some more info about your exception? If you're getting a Bad Request, you can catch it as a `BadRequestException`. Would you mind giving me both `exception.WebRequest.Url` and `((RestRequest)exception.WebRequest).Payload`?

**Comment by HurricanKai** [comment]
WebRequest.Url:
{https://discordapp.com/api/v7/users/@me/guilds}

payload:
Payload	"{\"limit\":100}"	string

i guess this goes with the fact that the RestOnlyClient gets 100 Guilds in his [Constructor. ](https://github.com/DSharpPlus/DSharpPlus/blob/80e68247d2fb2589fa353584d3ece1701efdef49/DSharpPlus.Rest/DiscordRestClient.cs#L29)
the Other Clients dont seem to do this Kind of stuff, only the BaseDiscordClient seems to Get the Current User in [InitializeAsync](https://github.com/DSharpPlus/DSharpPlus/blob/80e68247d2fb2589fa353584d3ece1701efdef49/DSharpPlus/BaseDiscordClient.cs#L110)

**Comment by nick-strohm** [comment]
Can confirm. Error occurres in this line https://github.com/DSharpPlus/DSharpPlus/blob/master/DSharpPlus.Rest/DiscordRestClient.cs#L29

**Comment by nick-strohm** [comment]
Testing fix 2b6bb3d58748d04bf064fc4b7cbb1a9645502bac in a second

**Comment by nick-strohm** [comment]
Well, now you can instantiate a RestClient successfully. But calling [InitializeCacheAsync](https://github.com/DSharpPlus/DSharpPlus/commit/2b6bb3d58748d04bf064fc4b7cbb1a9645502bac#diff-925a13aae597cd20f626c3823e76ae69R28) gives you the same exception.

The request and the code behind it looks good. Looks more like an issue on discords end:
The WebResponse only holds headers about cloudflare:
```
Connection: keep-alive
Server: cloudflare
Set-Cookie: __cfduid=XXX; expires=Mon, 11-Feb-19 13:37:46 GMT; path=/; domain=.discordapp.com; HttpOnly
Referrer-Policy: no-referrer
Expect-CT: max-age=604800, report-uri="https://report-uri.cloudflare.com/cdn-cgi/beacon/expect-ct"
CF-RAY: 3eb7b46b2cae2XXX-FRA
```
The response itself is just a html site saying:
```
400. That’s an error. 

Your client has issued a malformed or illegal request. That’s all we know. 
```

Maybe thats something for #devs?

**Comment by WamWooWam** [comment]
Well that's a Google error, so it could not even be hitting Discord?

**Comment by Emzi0767** [comment]
That's indeed a response from Google. Meaning this request gets caught before it reaches Discord. I'd make sure that it's sent with proper HTTP verb.

**Comment by nick-strohm** [comment]
Well, took a look at the response again. `WebRequest.Headers` is null

**Comment by NaamloosDT** [comment]
did you make sure to have the `guilds` oauth scope?

**Comment by NaamloosDT** [comment]
for caching guilds that is*

**Comment by NaamloosDT** [comment]
https://discordapp.com/developers/docs/resources/user#get-current-user-guilds

**Comment by NaamloosDT** [comment]
okay apparently my rest implementation is a half-done shitty implementation. gimme a sec to resolve this.

**Comment by HurricanKai** [comment]
I guess this can be closed with #245
